### PR TITLE
Consistent naming of 'tau' in predefined methods

### PR DESCRIPTION
**Pull request by timoML**
This PR introduces a consistent naming of tau in all predefined methods in order to avoid confusion between the tau definitions.
We now have in code and the pulsed gui x-axis:
- tau: as defined by the filter function of a dd-like experiment
- tau_pulse_spacing: physical spacing between MW pulses.

This PR changes the experiments Ramsey and Hahn echo to use the first definition. Thus, compared to before, "tau" is now corrected for the pi pulse duration there. **End-users should be informed about this change** prior to merging.

In addition, new cw dd methods have been contributed by @NicolasStaudenmaier.

## Description

Provides a general function pulse_objects::tau_2_pulse_spacing() that should be used to transform from tau (filter function definition) to the pulse spacing between pulses. Right now, this function just substracts the duration of a pi pulse and does some sanity checking. It can be extended to different transformations by providing a function pointer.

In detail changes to the predefined methods:
- Rabi: gui x-axis "tau" -> "tau_pulse_spacing"
- T1: gui x-axis "tau" -> "tau_pulse_spacing"
- Ramsey: **now uses tau** (filter function definition)
- Hahn echo: **now uses tau** (filter function definition)
- XY8: only code cleanup. 
- Advanced DD with KDD, etc.: can be found in internal iqo-sequences repo. They need some new code in sampling_functions::DDMethods that is included in this PR. Contributed by @genko8. 
- Continuous DD: newly introduced

## Motivation and Context
Avoid confusion between different definitions.

## How Has This Been Tested?
In dummy mode, all changed functions have been queried with tau values that might be problematic. Warning work as expected.
@anjusha-vs is currently testing the advanced dd sequences (iqo-sequences repo), but they are not relevant for this PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have documented my changes in the changelog (`documentation/changelog.md`)
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added/updated for the module the config example in the docstring of the class accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [x] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [x] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

**Comment by Neverhorst** [comment]
If I'm not mistaken, you now label the x-axis with something like "tau_spacing" but provide the x-values for the "ideal" tau, i.e. the DD-filter-tau.
This is certainly not intended and as far as I recall we decided to display and save to data the "ideal" tau (the one used for post-processing in most sequences).

**Comment by timoML** [comment]
Could you point to an exemplary sequence? That's definitely not intended. The tau_pulse_spacing label should only be used when there is no meaningful filter function (eg. T1).

**Comment by Neverhorst** [comment]
> Could you point to an exemplary sequence? That's definitely not intended. The tau_pulse_spacing label should only be used when there is no meaningful filter function (eg. T1).

You are right, I accidentally jumped generate methods while reading. My bad...

File: logic/pulsed/pulse_objects.py [hunk 1]
```diff
@@ -1099,6 +1100,51 @@ def sample_rate(self):
...
+                           custom_func=[None, None], **custom_kwwargs):
+        """
+        Converts tau to the physical pulse spacing between (microwave) pulses.
+        By definition, tau = 1/f where f is the filter frequency of a dynamical decoupling
+        experiment. For many cases this tau equals the time between the center of
+        consecutive pi pulses.
+        Thus, the default behavior is to subtract the duration of a pi pulse from tau.
+
+        :param t: tau (or tau_phys, if inverse==True) to be converted.
+        :param bool inverse: do the inverse transformation tau -> tau_phys
```
Comment (timoML) [inline logic/pulsed/pulse_objects.py:1114]: Tau_phys is a left-over, should be replaced with tau_pulse_spacing

**Comment by Neverhorst** [comment]
From a code point-of-view this looks good. It's a bit frustrating that git makes it so hard to refactor code while keeping attribution (blame).

**Review by Neverhorst — APPROVED** [review]
Great! Thanks everyone.

**Pull request merged**